cmd/mapcidr/main.go

```diff
@@ -374,12 +387,34 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
 
 	// Aggregate all ips into the minimal subset possible
 	if options.Aggregate {
-		cCidrsIPV4, cCidrsIPV6 := mapcidr.CoalesceCIDRs(allCidrs)
-		for _, cidrIPV4 := range cCidrsIPV4 {
-			outputchan <- cidrIPV4.String()
-		}
-		for _, cidrIPV6 := range cCidrsIPV6 {
-			outputchan <- cidrIPV6.String()
+		// Find out the possible CIDR prefixes from given IP address range
+		if options.CIDRFromIPRange {
+			// check if ipRange has more than 2 values
+			if len(ipRange)%2 == 1 {
+				gologger.Fatal().Msgf("IP range can not have more than 2 values.")
+			}
+			for i := 0; i < len(ipRange)-1; i = i + 2 {
```

Comment:
This loops is not required, if we need only single IP range?

Reply:
Removed the for loop